components/tabs/src/storage.rs

```diff
@@ -362,6 +378,32 @@ impl TabsStorage {
     }
 }
 
+// Trim the amount of tabs in a list to fit the specified memory size
+fn trim_tabs_length(
```

Comment:
I think this only ever practical for tabs, but if we happen to have a limit for history records/etc we might be able to make it generic over any `Vec<T>`, such that `T: Serialize` and have it a part of `sync15`

But since Tabs is the only one that we might hit limit with I like this here not to overengineer!

Reply:
That's an interesting idea! One slight issue with this would be that the `Vec<>` in question might not be the entire payload - eg, for history, it would be the `visits` element which gets trimmed. I *think* this could still be expressed though, but agree we should wait for an actual other use-case.